packages/@webex/internal-plugin-llm/src/llm.ts

```diff
@@ -75,53 +73,62 @@ export default class LLMChannel extends (Mercury as any) implements ILLMChannel
    * @param {string} llmSocketUrl
    * @returns {Promise<void>}
    */
-  private register = (llmSocketUrl:string):Promise<void> => this.request({
-    method: 'POST',
-    url: llmSocketUrl,
-  }).then((res: { body: { webSocketUrl: string; binding: string; }; }) => {
-    this.webSocketUrl = res.body.webSocketUrl;
-    this.binding = res.body.binding;
-  }).catch((error:any) => {
-    LoggerProxy.logger.error(`Error connecting to websocket: ${error}`);
-    throw error;
-  });
+  private register = (llmSocketUrl: string): Promise<void> =>
+    this.request({
+      method: 'POST',
+      url: llmSocketUrl,
+    })
+      .then((res: {body: {webSocketUrl: string; binding: string}}) => {
+        this.webSocketUrl = res.body.webSocketUrl;
+        this.binding = res.body.binding;
+      })
+      .catch((error: any) => {
+        LoggerProxy.logger.error(`Error connecting to websocket: ${error}`);
+        throw error;
+      });
 
   /**
    * Register and connect to the websocket
    * @param {string} locusUrl
    * @param {string} datachannelUrl
    * @returns {Promise<void>}
    */
-  public registerAndConnect = (locusUrl:string, datachannelUrl:string):Promise<void> => this.register(datachannelUrl).then(() => {
-    if (!locusUrl || !datachannelUrl) return undefined;
-    this.locusUrl = locusUrl;
-    this.connect(this.webSocketUrl).then(() => {
-      this.registered = true;
+  public registerAndConnect = (locusUrl: string, datachannelUrl: string): Promise<void> =>
+    this.register(datachannelUrl).then(() => {
+      if (!locusUrl || !datachannelUrl) return undefined;
+      this.locusUrl = locusUrl;
+      this.connect(this.webSocketUrl).then(() => {
+        this.registered = true;
+      });
     });
-  });
 
   /**
    * Tells if LLM socket is connected
    * @returns {boolean} isRegistered
    */
-  public isConnected = ():boolean => this.registered;
+  public isConnected = (): boolean => this.registered;
 
   /**
    * Tells if LLM socket is connected
    * @returns {bool} isRegistered
    */
-  public getBinding = ():string => this.binding;
+  public getBinding = (): string => this.binding;
 
   /**
    * Get Locus URL for the connection
    * @returns {string} locus Url
    */
-  public getLocusUrl = ():string => this.locusUrl;
+  public getLocusUrl = (): string => this.locusUrl;
 
   /**
    * Disconnects websocket connection
    * @returns {Promise<void>}
    */
-  public disconnectLLM = (): Promise<void> => this.disconnect();
+  public disconnectLLM = (): Promise<void> =>
+    this.disconnect().then(() => {
+      this.registered = false;
+      this.locusUrl = undefined;
```

Comment:
the type for `this.locusUrl` is defined as `string` so aren't you getting eslint warnings here? shouldn't the type be defined as `private locusUrl?: string;`
same applies to other fields being reset here

Reply:
modified